src/Handler/CurlFactory.php

```diff
@@ -359,8 +359,14 @@ private function applyHandlerOptions(EasyHandle $easy, array &$conf): void
                     }
                     // If it's a directory or a link to a directory use CURLOPT_CAPATH.
                     // If not, it's probably a file, or a link to a file, so use CURLOPT_CAINFO.
-                    if (\is_dir($options['verify']) ||
-                        (\is_link($options['verify']) && \is_dir(\readlink($options['verify'])))) {
+                    if (
+                        \is_dir($options['verify']) ||
+                        (
+                            \is_link($options['verify']) === true &&
+                            \readlink($options['verify']) !== false &&
+                            \is_dir(\readlink($options['verify']))
+                        )
+                    ) {
```

Comment:
This was flagged by phpstan:
```
 ------ ----------------------------------------------------------- 
  Line   Handler/CurlFactory.php                                    
 ------ ----------------------------------------------------------- 
  363    Parameter #1 $filename of function is_dir expects string,  
         string|false given.                                        
 ------ ----------------------------------------------------------- 
```
I can alternatively ignore this?

Reply:
Don't call readlink over and over. Assign it to a variable, then you will avoid both slow code and warnings.

Reply:
I've now squashed that into 1 commit 0c2f920

Reply:
This fix has been applied on master, so no longer on this PR